### PR TITLE
added first route in skill test routes

### LINKED ISSUE #141: Add Skill Test routes
Add `Skill Test` routes in README.md. 

Refer [this PR](https://github.com/CodeTrophs/Backend/pull/136) for details that should be included for different routes added.

### PR DESCRIPTION
**Pull request by ankitvashisht12**
Fixes #141

### CODE REVIEW

File: docs/skill-tests.md [hunk 1]
```diff
@@ -1 +1,32 @@
-#### This is the beginning of the skill tests routes docs!
+### Please find the different routes available for `Skill Test` below -
+
+##### **Skill Test routes**
+
+1. `GET [PROTECTED ROUTE] /v1/skillTest` - This route lists all the skill Tests.
```
Comment (Dishebh) [inline docs/skill-tests.md:5]: `ADMIN PROTECTED ROUTE`

File: docs/skill-tests.md [hunk 2]
```diff
@@ -1 +1,32 @@
...
+       }
+     }
+     ```
+   - Status code of the response - `200`
+   - Query parameters in request (default) -
+     ```json
+     {
+       "page": 1,
+	   "per_page": 10,
+	   "isPublished": "BOOL VALUE"
```
Comment (Dishebh) [inline docs/skill-tests.md:30]: Extra tab-width here

**Comment by Dishebh** [comment]
Add all remaining routes of skill tests in this pr itself

**Comment by ankitvashisht12** [comment]
Okay

File: docs/skill-tests.md [hunk 3]
```diff
@@ -1 +1,247 @@
...
+        "testId": "TEST_ID"
+      }
+      ```
+    
+
+3. `POST [ADMIN PROTECTED ROUTE] /v1/skillTest` - This route lets user post skill Test.
+    - Request body -
+      ```json
+      {
+        "name": "USER NAME",
```
Comment (Dishebh) [inline docs/skill-tests.md:73]:
```suggestion
        "name": "SKILL TEST NAME",
```

File: docs/skill-tests.md [hunk 4]
```diff
@@ -1 +1,247 @@
...
+        "image": "IMAGE URL",
+        "description": "DESCRIPTION"
+      }
+      ```
+    - Response body -
+      ```json
+      {
+        "data": {
+          "_id": "ID",
+          "name": "USER NAME",
```
Comment (Dishebh) [inline docs/skill-tests.md:83]:
```suggestion
          "name": "SKILL TEST NAME",
```

File: docs/skill-tests.md [hunk 5]
```diff
@@ -1 +1,247 @@
...
+      ```
+    - Response body -
+      ```
+      Skill Test published successfully
+      ```
+    - Status code of the response - `200`
+    - Route Parameters in request URL -
+      ```json
+      {
+      "testId": "TEST_ID"
```
Comment (Dishebh) [inline docs/skill-tests.md:106]:
```suggestion
        "testId": "TEST_ID"
```

File: docs/skill-tests.md [hunk 6]
```diff
@@ -1 +1,247 @@
...
+          "correctIndex": "CORRECT INDEX",
+          "testId": "TEST_ID"
+        }
+      }
+      ```
+    - Status code of the response - `200`
+    - Route Parameters in request URL -
+      ```json
+      {
+      "testId": "TEST_ID"
```
Comment (Dishebh) [inline docs/skill-tests.md:135]:
```suggestion
        "testId": "TEST_ID"
```

File: docs/skill-tests.md [hunk 7]
```diff
@@ -1 +1,247 @@
...
+            "options": "OPTIONS",
+            "correctIndex": "CORRECT INDEX",
+            "testId": "TEST_ID"
+          }
+        }
+        ```
+    - Route Parameters in request URL -
+      ```json
+      {
+      "questionId": "QUESTION_ID"
```
Comment (Dishebh) [inline docs/skill-tests.md:168]:
```suggestion
        "questionId": "QUESTION_ID"
```

File: docs/skill-tests.md [hunk 8]
```diff
@@ -1 +1,247 @@
...
+      }
+      ```
+    
+7. `PATCH [ADMIN PROTECTED ROUTE] /v1/skillTest/:id` - This route lets user update the skill Test of particular skill test id.
+    - Request body -
+      ```json
+      {}
```
Comment (Dishebh) [inline docs/skill-tests.md:175]:
```suggestion
      {
        "name",
        "image",
        "description"
      }
```

File: docs/skill-tests.md [hunk 9]
```diff
@@ -1 +1,251 @@
-#### This is the beginning of the skill tests routes docs!
+### Please find the different routes available for `Skill Test` below -
+
+##### **Skill Test routes**
+
+1. `GET [USER PROTECTED ROUTE] /v1/skillTest` - This route lists all the skill Tests.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:9]:
```suggestion
```

File: docs/skill-tests.md [hunk 10]
```diff
@@ -1 +1,251 @@
...
+      {
+        "testId": "TEST_ID"
+      }
+      ```
+
+10. `DELETE [ADMIN PROTECTED ROUTE] /v1/skillTest/:id` - This route deletes skill test for particular skill test id.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:240]:
```suggestion
```

File: docs/skill-tests.md [hunk 11]
```diff
@@ -1 +1,251 @@
...
+        "questionId": "QUESTION_ID"
+      }
+      ```
+    
+
+9. `DELETE [ADMIN PROTECTED ROUTE] /v1/skillTest/publish/:testId` - This route deletes published test for particular test id.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:223]:
```suggestion
```

File: docs/skill-tests.md [hunk 12]
```diff
@@ -1 +1,251 @@
...
+      {
+        "id": "ID"
+      }
+      ```
+
+8. `DELETE [ADMIN PROTECTED ROUTE] /v1/skillTest/question/:questionId` - This route deletes question for particular question id.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:205]:
```suggestion
```

File: docs/skill-tests.md [hunk 13]
```diff
@@ -1 +1,251 @@
...
+      }
+      ```
+    - Status code of the response - `200`
+    
+
+4. `PATCH [ADMIN PROTECTED ROUTE] /v1/skillTest/publish/:testId` - This route lets user publish skill test.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:97]:
```suggestion
```

File: docs/skill-tests.md [hunk 14]
```diff
@@ -1 +1,251 @@
...
+        "per_page": 10,
+        "isPublished": "BOOL VALUE"
+      }
+      ```
+
+2. `GET [USER PROTECTED ROUTE] /v1/skillTest/:testId` - This route lists all the skill Test Questions of particular test id.
+    - Request body -
+      ```json
+      {}
+      ```
```
Comment (Dishebh) [inline docs/skill-tests.md:38]:
```suggestion
```

**Review by Dishebh — APPROVED** [review]

**Pull request merged**